kernel/src/engine/default/mod.rs

```diff
@@ -108,6 +113,32 @@ impl<E: TaskExecutor> DefaultEngine<E> {
     pub fn get_object_store_for_url(&self, _url: &Url) -> Option<Arc<DynObjectStore>> {
         Some(self.store.clone())
     }
+
+    pub async fn write_parquet(
+        &self,
+        data: &ArrowEngineData,
+        write_context: &WriteContext,
+        partition_values: HashMap<String, String>,
+        data_change: bool,
+    ) -> DeltaResult<Box<dyn EngineData>> {
+        let transform = write_context.logical_to_physical();
+        let input_schema: Schema = data.record_batch().schema().try_into()?;
+        let output_schema = write_context.schema();
+        let logical_to_physical_expr = self.get_expression_handler().get_evaluator(
```

Comment:
Ideally we should be creating the expression evaluator just once? 
Or do we worry the record batch schemas will somehow vary in e.g. field order?

Reply:
well for now, writing single parquet files aren't we bound to the input schema? That is, for each input data we have (possibly) different input schema which then means different `logical_to_physical_expr`?

Reply:
Partition values could be one, I guess. Normally a partition-aware engine would want to pass us the partition value literals and let us decide whether to embed them in the outgoing data? And I guess some future table feature could conceivably require per-file awareness?